packages/jest-haste-map/src/crawlers/node.js

```diff
@@ -30,7 +30,10 @@ function find(
     activeCalls++;
     fs.readdir(directory, (err, names) => {
       activeCalls--;
-
+      if (!names) {
```

Comment:
How about going a bit more idiomatic Node.js with `if (err) {`?

Reply:
Yeah, if that's what happens (unhandled error) that's the correct fix. @emuraton do you have a reproduction case for your error?

Reply:
@SimenB I did not suceed to replicate the issue case of @timbault.
But from the stack trace, callback doesnt return an error @thymikee, just `names` undefined.
I can add both check if you think its better (`if( err || !names)`)

Doest it make sense?

Reply:
If it returns undefined names, but null err, it's a bug in Node imho and you should also file it there

Reply:
Agreed, the docs says you should always get an array: https://nodejs.org/api/fs.html#fs_fs_readdir_path_options_callback

Reply:
Sorry for the late answer, after double thought I think you were right before actually. Checking `err` should be enough.
If I find a bit of time today, I'll try to replicate to see if its a bug from Node or if your first comments were right

Reply:
Thanks, looking forward to it!